src/Chainweb/Pact/PactService/Pact4/ExecBlock.hs

```diff
@@ -253,11 +254,23 @@ validateParsedChainwebTx logger v cid dbEnv txValidationTime bh tx
   | otherwise = do
       checkUnique logger dbEnv tx
       checkTxHash logger v cid bh tx
+      checkChain cid tx
       checkTxSigs logger v cid bh tx
       checkTimes logger v cid bh txValidationTime tx
       _ <- checkCompile logger v cid bh tx
       return ()
 
+checkChain
+  :: ChainId
+  -> Pact4.Command (Pact4.PayloadWithText Pact4.PublicMeta code)
+  -> ExceptT InsertError IO ()
+checkChain cid
+  (view (Pact4.cmdPayload . to Pact4.payloadObj . Pact4.pMeta . Pact4.pmChainId) -> txCid)
+  | Pact4.assertChainId cid txCid =
+    return ()
```

Comment:
👍

Reply:
Seems like this whole thing could be

```haskell
unless (Pact4.assertChainId cid txCid) $ 
  throwError $ InsertErrorWrongChain (chainIdToText cid (Pact4._chainId txCid)
```

and then we don't need guard clauses that make it seem like we're doing something more than just checking that bool.